a10_octavia/controller/worker/controller_worker.py

```diff
@@ -190,7 +191,11 @@ def create_listener(self, listener_id):
             raise db_exceptions.NoResultFound
 
         load_balancer = listener.load_balancer
-        if listener.project_id in CONF.hardware_thunder.devices:
+
+        if any([any([proj in utils.get_parent_project_list() for proj in (listener.project_id,
```

Comment:
This is some pretty messy list comprehension. Splitting it up a bit would make the section far more readable for a similar performance and LOC cost.